core/stream/convert.go

```diff
@@ -373,13 +373,34 @@ func ftou(count int, dst, src buf) error {
 	dstBits, srcBits := dstTy.Bits(), srcTy.Bits()
 	dstMask := (1 << dstBits) - 1
 	ss := binary.BitStream{Data: src.bytes, ReadPos: src.offset}
+	ss2 := binary.BitStream{Data: src.bytes, ReadPos: src.offset}
 	ds := binary.BitStream{Data: dst.bytes, WritePos: dst.offset}
 	switch {
 	case srcIsF16:
 		scale := float32(dstMask)
+		min, max := float32(math.MaxFloat32), float32(0)
+		for i := 0; i < count; i++ {
+			f := f16.Number(ss2.Read(16)).Float32()
+			if f < min {
+				min = f
+			}
+			if f > max {
+				max = f
+			}
+			ss2.ReadPos += src.stride - 16
+		}
 		for i := 0; i < count; i++ {
 			f := f16.Number(ss.Read(16)).Float32()
 			if norm {
+				if src.component.Channel == Channel_Depth {
+					f = (f - min) * (1 / (max - min))
+				} else if min < 0 {
+					if f <= 0 {
+						f = (f - min) * (0.5 / -min)
+					} else {
+						f = f*(0.5/max) + 0.5
```

Comment:
possible division by zero

Reply:
Nope. If max were zero, we would never have a positive f

Reply:
Stability concern?